internal/ignition/disconnected.go

```diff
@@ -233,3 +301,67 @@ func generateUnconfiguredIgnition(executer executer.Executer, releasePath string
 
 	return string(ignitionContent), nil
 }
+
+func createNMStateConfigManifests(infraEnv *common.InfraEnv, manifestsDir string, log logrus.FieldLogger) error {
+	if infraEnv.StaticNetworkConfig == "" {
+		log.Debug("No static network configuration present, skipping NMStateConfig manifests")
+		return nil
+	}
+
+	var staticNetworkConfigs []*models.HostStaticNetworkConfig
+	if err := json.Unmarshal([]byte(infraEnv.StaticNetworkConfig), &staticNetworkConfigs); err != nil {
+		return errors.Wrap(err, "failed to unmarshal static network config")
+	}
+
+	for i, config := range staticNetworkConfigs {
+		if config == nil || config.NetworkYaml == "" {
+			continue
+		}
+
+		// Convert mac_interface_map to v1beta1.Interface format
+		interfaces := make([]*v1beta1.Interface, 0, len(config.MacInterfaceMap))
+		for _, macInterface := range config.MacInterfaceMap {
+			if macInterface.MacAddress != "" && macInterface.LogicalNicName != "" {
+				interfaces = append(interfaces, &v1beta1.Interface{
+					Name:       macInterface.LogicalNicName,
+					MacAddress: macInterface.MacAddress,
+				})
+			}
+		}
+
+		if len(interfaces) == 0 {
+			log.Warnf("Skipping static network config %d: no valid interfaces found", i)
+			continue
+		}
+
+		nmStateConfig := &v1beta1.NMStateConfig{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "agent-install.openshift.io/v1beta1",
```

Comment:
If its specified, it is better to use like https://github.com/openshift/installer/blob/main/pkg/asset/agent/manifests/nmstateconfig.go#L103-L105